components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2RegistrationEventPayloadBuilder.java

```diff
@@ -144,4 +161,67 @@ private List<UserClaim> filterUserClaimsForUserAdd(Map<String, String> userClaim
         return userClaimList;
     }
 
+    @Override
+    public EventPayload buildRegistrationFailureEvent(EventData eventData) throws IdentityEventException {
+
+        Map<String, Object> properties = eventData.getEventParams();
+        String tenantId = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_ID));
+        String tenantDomain = String.valueOf(properties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN));
+
+        String userStoreDomainName = resolveUserStoreDomain(properties);
+        UserStore userStore = null;
+
+        if (StringUtils.isNotBlank(userStoreDomainName)) {
+            userStore = new UserStore(userStoreDomainName);
+        }
+
+        User newUser = new User();
+        enrichUser(properties, newUser);
+
+        Organization organization = new Organization(tenantId, tenantDomain);
+        Flow flow = IdentityContext.getThreadLocalIdentityContext().getFlow();
+        String initiatorType = null;
+        String action = null;
+        if (flow != null) {
+            initiatorType = flow.getInitiatingPersona().name();
+            action = flow.getName().name();
```

Comment:
This action should not be the internal flow name. It should be something like 
REGISTER -> in self registration or admin onboard
INVITE -> in an admin invite flow